### PR TITLE
[#111731500] Updated AzureRm module to include major versions for each package

### PR DESCRIPTION
**Pull request by hovsepm**

### CODE REVIEW

**Comment by hovsepm** [comment]
sign job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/210/

File: tools/AzureRM/AzureRM.psm1 [hunk 1]
```diff
@@ -1,33 +1,36 @@
...
-  "AzureRM.Sql",
-  "AzureRM.Storage",
-  "AzureRM.StreamAnalytics",
-  "AzureRM.Tags",
-  "AzureRM.TrafficManager",
-  "AzureRM.UsageAggregates",
-  "AzureRM.Websites"
-)
+$AzureRMDependencies = @{
+  "Azure.Storage" = "1.0";
```
Comment (markcowl) [inline tools/AzureRM/AzureRM.psm1]: This should only contain major version

File: tools/AzureRM/AzureRM.psm1 [hunk 2]
```diff
@@ -64,10 +67,10 @@ function CheckIncompatibleVersion([bool]$Force)
   }
 }
 
-function Install-ModuleWithVersionCheck([string]$Name,[string]$MajorVersion,[string]$Repository,[string]$Scope)
+function Install-ModuleWithVersionCheck([string]$Name,[string]$MinimumVersion,[string]$Repository,[string]$Scope)
 {
-  $_MinVer = "$MajorVersion.0.0.0"
-  $_MaxVer = "$MajorVersion.9999.9999.9999"
+  $_MinVer = $MinimumVersion
+  $_MaxVer = "$($_MinVer.Split(".")[0]).9999.0"
```
Comment (markcowl) [inline tools/AzureRM/AzureRM.psm1]: major version from the file, only

File: tools/AzureRM/AzureRM.psm1 [hunk 3]
```diff
@@ -162,14 +170,17 @@ function Import-AzureRM
   param(
   [Parameter(Position=0, Mandatory = $false)]
   [string]
-  $MajorVersion = $AzureMajorVersion)
+  $MinorVersion)
```
Comment (markcowl) [inline tools/AzureRM/AzureRM.psm1]: No reason to change the parameter name - it actually is the major version

File: tools/AzureRM/AzureRM.psm1 [hunk 4]
```diff
@@ -111,7 +114,7 @@ function Update-AzureRM
   param(
   [Parameter(Position=0, Mandatory = $false)]
   [string]
-  $MajorVersion = $AzureMajorVersion,
+  $MinVersion,
```
Comment (markcowl) [inline tools/AzureRM/AzureRM.psm1]: I would leave the parameter name as is

File: tools/AzureRM/AzureRM.psm1-help.xml [hunk 5]
```diff
@@ -305,7 +281,19 @@
 		</command:syntaxItem>
 	</command:syntax>
 	<command:parameters>
-		<command:parameter required="false" variableLength="false" globbing="false" pipelineInput="false" position="0">
+    <command:parameter required="false" variableLength="false" globbing="false" pipelineInput="false" position="0">
+      <maml:name>MinVersion</maml:name>
```
Comment (markcowl) [inline tools/AzureRM/AzureRM.psm1-help.xml]: same comment

**Pull request merged**